code/game/machinery/abominationmachine.dm

```diff
@@ -0,0 +1,179 @@
+				update_icon()
+			else
+				n_organs_eaten = n_organs_eaten + 1
+			qdel(thing)
+			return TRUE	
+			
+		if (!has_creature)//no creature inside
+			return FALSE
+		
+		if (istype(thing, obj/item/organ/brain))
+		
+			to_chat(user, "<span class='notice'>You offer [src] to [SM]...</span>")	
+			var/obj/item/organ/brain/brian = thing
+			
+			if(brian.brainmob && brian.brainmob.mind) //enslave the current mind into the creature
+			
+				var/mob/creature = release_creature(FALSE)
+				brainmob.mind.transfer_to(creature)
+				creature.key = brian.brainmob.mind.key
+				creature.mind.enslave_mind_to_creator(user)
+				creature.sentience_act()
+				to_chat(creature, "<span class='warning'>All at once it makes sense: you know what you are and who you are! Self awareness is yours!</span>")
+				to_chat(creature, "<span class='userdanger'>You are grateful to be self aware and owe [user.real_name] a great debt. Serve [user.real_name], and assist [user.p_them()] in completing [user.p_their()] goals at any cost.</span>")
+				creature.copy_languages(user)	
+			
+			else //create a ghostpoll
+				var/list/candidates = pollCandidatesForMob("Do you want to play as [SM.name]?", ROLE_SENTIENCE, null, ROLE_SENTIENCE, 50, SM, POLL_IGNORE_SENTIENCE_POTION)
+				if(LAZYLEN(candidates))
+				
+					var/mob/creature = release_creature(FALSE)
+					var/mob/dead/observer/C = pick(candidates)
+					creature.key = C.key
+					creature.mind.enslave_mind_to_creator(user)
+					creature.sentience_act()
+					to_chat(creature, "<span class='warning'>All at once it makes sense: you know what you are and who you are! Self awareness is yours!</span>")
+					to_chat(creature, "<span class='userdanger'>You are grateful to be self aware and owe [user.real_name] a great debt. Serve [user.real_name], and assist [user.p_them()] in completing [user.p_their()] goals at any cost.</span>")
+					creature.copy_languages(user)							
+		else
+			n_organs_eaten = n_organs_eaten + 1
+			qdel(thing)
+			return TRUE		
+	return FALSE
+	
+/obj/machinery/abomachine/interact(mob/user)
+	to_chat(src, "<span class='warning'>You start opening the [src]</span>")
+	if(do_after(user, 40, src))
+		var/monster = release_creature(FALSE)		
+		log_game("[user] released [monster]")	//no shittery
+				
+/obj/machinery/abomachine/attackby(obj/item/I, mob/user, params)
+	if (istype(I, obj/item/bodypart))
+		if (!feed_part(I))
+			//you cant attack this body part to the abomination
+			to_chat(user, "<span class='notice'>[I] cannot attach on your abomination.</span>")
+		else
+			if (!has_creature)
+				to_chat(user, "<span class='notice'>There is no creature inside of the [src].</span>")
+			else
+				to_chat(user, "<span class='notice'>You feed the [I] through a small gap in [src], as it attaches to the fleshy mass.</span>")
+				
+	if(default_unfasten_wrench(user, I))
+		return
+	
+	if (istype(I, /obj/item/pen))
+		var/input = stripped_input(user,"What will your horror be named?", ,"", MAX_NAME_LEN)
```

Comment:
Change to `stripped_input(user, "What will your horror be named?", max_length=MAX_NAME_LEN)`